content/javascript/concepts/strings/terms/substr/substr.md

```diff
@@ -0,0 +1,69 @@
+---
+Title: '.substr()'
+Description: 'Extracts a portion of a string, starting from a specified position up to a given number of characters.'
+Subjects:
+  - 'Web Development'
+  - 'Computer Science'
+Tags:
+  - 'Strings'
+  - 'Methods'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/front-end-engineer-career-path'
+---
+
+The **`.substr()`** method in JavaScript extracts a portion of a string, starting from a specified index position up to a given number of characters. The index count begins with zero.
+
+## Syntax
+
+```pseudo
+string.substr(startingIndex, length);
+```
+
+- `startingIndex`: determines the index from which the substring will begin.
+- `length`: (optional) is the number of characters to extract.
+
+If `length` is not provided, `.substr()` will extract the rest of the characters until the end of the string.
+
+## Example 1
+
+Extracting a portion of a string with a specified `length`.
+
+```js
+console.log('The Intro to JavaScript is fun to learn.'.substr(4, 19));
+// Output: Intro to JavaScript
+```
+
+## Example 2
+
+Extracting a portion of a string _without_ a specified `length`.
+
+```js
+console.log('The Intro to JavaScript is fun to learn.'.substr(4));
+// Output: Intro to JavaScript is fun to learn.
+```
+
+The `.substr()` can also extract from the end of the string when provided with a negative starting index position.
+
+## Example 3
+
+Extracting from the end of a string.
+
+```js
+console.log('The Intro to JavaScript is fun to learn.'.substr(-27, 10));
+// Output: JavaScript
+```
```

Comment:
Let's have all the 3 examples with different test cases in one, as follows:

## Example
```js
// Extracting a portion of a string with a specified length

var sentence1 = "The Intro to JavaScript is fun to learn."
console.log(sentence1.substr(4, 19));

// Extracting a portion of a string without a specified length
var sentence2 = "The Intro to JavaScript is fun to learn"
console.log(sentence2.substr(4));
```

Add all the 3 use cases (examples) in one block, you can use comments `//` to explain the use case in the code. 
For displaying the output, use the ```shell``` block after the ##Example block.